plugins/Annotations/API.php

```diff
@@ -188,17 +216,27 @@ public function get(int $idSite, int $idNote): array
      *                 ),
      *                 8 => array(...)
      *               )
+     * @throws Exception when user is not a superuser
      */
     public function getAll(string $idSite, ?string $date = null, string $period = 'day', ?int $lastN = null): array
     {
         Piwik::checkUserHasViewAccess($idSite);
 
-        $annotations = new AnnotationList($idSite);
-
-        // if date/period are supplied, determine start/end date for search
-        list($startDate, $endDate) = Annotations::getDateRangeForPeriod($date ?? false, $period, $lastN ?? false);
+        $ids = array_map('intval', Site::getIdSitesFromIdSitesString($idSite));
+        $model = new Model();
+        $annotations = [];
+        foreach ($ids as $id) {
+            [$startDate, $endDate] = Annotations::getDateRangeForPeriod($date, $period, $lastN ?? false, $id);
+            if (!($startDate && $endDate)) {
+                continue;
+            }
```

Comment:
Both here and in `getAnnotationCountForDates` the behaviour changed from "give me everything if I don't pass a date" to "silently return nothing".

Sure you can just pass a huge date range, but it is breaking the API for those using it without a date.